File: app/src/main/java/net/numa08/tiqavapp/service/LoadTiqavService.java
package net.numa08.tiqavapp.service;

import android.util.Log;

import net.numa08.tiqa4k.Tiqav;
import net.numa08.tiqa4k.TiqavService;
import net.numa08.tiqavapp.TiqavApplication;
import net.numa08.tiqavapp.realm.configurator.CacheRealmConfigurator;

import org.androidannotations.annotations.EIntentService;
import org.androidannotations.annotations.ServiceAction;
import org.androidannotations.api.support.app.AbstractIntentService;

import java.util.Arrays;

import javax.inject.Inject;

import io.realm.Realm;
import io.realm.RealmConfiguration;
import rx.Observer;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

@EIntentService
public class LoadTiqavService extends AbstractIntentService{

    @Inject
    TiqavService tiqavService;
    @Inject
    CacheRealmConfigurator cacheRealmConfigurator;


    public LoadTiqavService() {
        super("LoadTiqavService");
    }

    @Override
    public void onCreate() {
        super.onCreate();
        TiqavApplication.getInstance().getComponent().inject(this);
    }

    @ServiceAction
    void loadNewest() {
        tiqavService
                .newest()
                .subscribe(new Observer<Tiqav[]>() {
                    @Override
                    public void onCompleted() {
                        Log.d("tiqav", "complete TiqavService:newest");
                    }

                    @Override
                    public void onError(Throwable e) {
                        Log.d("tiqav", "cache error on TiqavService:newest");
                    }

                    @Override
                    public void onNext(Tiqav[] tiqavs) {
                        final Realm realm = Realm.getInstance(cacheRealmConfigurator.getRealmConfiguration());
                        realm.executeTransaction(r -> r.copyToRealmOrUpdate(Arrays.asList(tiqavs)));
                        realm.close();
                    }
                });
    }
}

File: app/src/main/java/net/numa08/tiqavapp/service/LoadTiqavService.kt
package net.numa08.tiqavapp.service

import android.app.IntentService
import android.content.Intent
import android.util.Log
import io.realm.Realm
import net.numa08.tiqa4k.Tiqav
import net.numa08.tiqa4k.TiqavServiceProperty
import net.numa08.tiqavapp.realmConfigurator
import rx.Observer
import rx.schedulers.Schedulers

class LoadTiqavService : IntentService("LoadTiqavService") {

    companion object {
        val ACTION_LOAD = LoadTiqavService::class.java.simpleName + "ACTION_LOAD"
    }

    var tiqavService by TiqavServiceProperty()
    val realmConfig by realmConfigurator()

    var observer = object :Observer<Array<Tiqav>> {
        override fun onCompleted() {}

        override fun onNext(t: Array<Tiqav>?) {
            t?.let { t ->
                Realm.getInstance(realmConfig).use { r ->
                    r.executeTransaction { r ->
                        r.copyToRealmOrUpdate(t.asList())
                    }
                }
            }
        }

        override fun onError(e: Throwable?) {
            Log.e("tiqav", "fail fetch tiqav ", e)
        }

    }

    override public fun onHandleIntent(intent: Intent?) {
        intent?.let { i ->
            if(i.action == ACTION_LOAD) {
                loadNewest()
            }
        }
    }

    fun loadNewest() {
        tiqavService
            .newest()
            .subscribeOn(Schedulers.newThread())
            .subscribe(observer)
    }

}